app/src/tx_reviewer.rs

```diff
@@ -0,0 +1,594 @@
+        assert!(prefix.len() + 3 <= 13);
+        output[..prefix.len()].copy_from_slice(prefix);
+        let num_str_bytes = I32::unsafe_from(index).to_str(&mut output[prefix.len()..]);
+        if num_str_bytes.is_none() {
+            return Err(ErrorCode::Overflow);
+        }
+        let total_size = prefix.len() + num_str_bytes.unwrap().len();
+        self.write_str(&output[..total_size])
+    }
+
+    pub fn write_address(&mut self, prefix: u8, hash: &[u8; 32]) -> Result<usize, ErrorCode> {
+        let mut output = [0u8; 46];
+        let str_bytes = to_base58_address(prefix, hash, &mut output)?;
+        self.write_str(str_bytes)
+    }
+
+    fn prepare_output(
+        &mut self,
+        output: &AssetOutput,
+        current_index: usize,
+        temp_data: &[u8],
+    ) -> Result<OutputIndexes, ErrorCode> {
+        assert!(self.index == 0);
+        let review_message_from_index = self.index;
+        let review_message_to_index = self.write_index_with_prefix(current_index, b"Output #")?;
+
+        let alph_amount_from_index = self.index;
+        let alph_amount_to_index = self.write_alph_amount(&output.amount)?;
+
+        let address_from_index = self.index;
+        let address_to_index = match &output.lockup_script {
+            LockupScript::P2PKH(hash) | LockupScript::P2SH(hash) => {
+                self.write_address(output.lockup_script.get_type(), &hash.0)?
+            }
+            LockupScript::P2MPKH(_) => self.write_multi_sig(temp_data)?,
+            _ => panic!(), // dead branch
+        };
+
+        let output_indexes = OutputIndexes {
+            review_message: (review_message_from_index, review_message_to_index),
+            alph_amount: (alph_amount_from_index, alph_amount_to_index),
+            address: (address_from_index, address_to_index),
+            token: None,
+        };
+        if output.tokens.is_empty() {
+            return Ok(output_indexes);
+        }
+
+        // Asset output has at most one token
+        let token = output.tokens.get_current_item().unwrap();
+        let token_id_from_index = self.index;
+        let token_id_to_address = self.write_token_id(&token.id)?;
+
+        let token_amount_from_index = self.index;
+        let token_amount_to_index = self.write_token_amount(&token.amount)?;
+
+        Ok(OutputIndexes {
+            token: Some(TokenIndexes {
+                token_id: (token_id_from_index, token_id_to_address),
+                token_amount: (token_amount_from_index, token_amount_to_index),
+            }),
+            ..output_indexes
+        })
+    }
+
+    fn get_str_from_range(&self, range: (usize, usize)) -> Result<&str, ErrorCode> {
+        let bytes = self.read_from_range(range.0, range.1);
+        bytes_to_string(bytes)
+    }
+
+    pub fn review_network(id: u8) -> Result<(), ErrorCode> {
+        let network_type = match id {
+            0 => "mainnet",
+            1 => "testnet",
+            _ => "devnet",
```

Comment:
Let's use this: `0 -> mainnet, 1 -> testnet, 3..4 -> devnet, n -> ID: n`

Reply:
Fixed in this commit: https://github.com/alephium/ledger-alephium/pull/17/commits/976928ef469ff936dc8ee7e8db61a5d600ce1f56